modules/iknowpy/iknowpy/engine.pyx

```diff
@@ -41,75 +43,113 @@ cdef char* aType_to_str(Attribute t) except NULL:
 	raise ValueError('Attribute type {} is unrecognized.'.format(t))
 
 
-cdef class UserDictionary:
+Entry = namedtuple("Entry", ["literal", "label"])
+
+cdef class UserDictionary(object):
 	"""A class that represents a user dictionary"""
 	cdef CPPUserDictionary user_dictionary
 
-	def __cinit__(self):
-		"""Initialize the underlying C++ iKnowUserDictioanry class"""
+	# keep track of entries in a readable form
+	cdef public entries
+
+	def __cinit__(self, load_entries=[]):
+		"""Initialize the underlying C++ iKnowUserDictionary class"""
 		self.user_dictionary = CPPUserDictionary()
 
+	def __init__(self, load_entries=[]):
```

Comment:
It's usually bad practice to use a mutable value such as a list as a default argument because it is initialized the first time this function is called, and the same list is re-used in future invocations of this function. This is a quirk of Python. The typical way to get around this is to use `load_entries=None` as the default argument and do

```
if load_entries is None:
    load_entries = []
```

within the function.